src/library_pthread.js

```diff
@@ -494,9 +495,9 @@ var LibraryPThread = {
   $cleanupThread: function(pthread_ptr) {
     if (ENVIRONMENT_IS_PTHREAD) throw 'Internal Error! cleanupThread() can only ever be called from main application thread!';
     if (!pthread_ptr) throw 'Internal Error! Null pthread_ptr in cleanupThread!';
```

Comment:
These two lines should probably be in `#if ASSERTIONS` shouldn't that?  Although unrelated to this change.

Reply:
Hard to tell that, as `$killThread`, `$cancelThread` and `$spawnThread` functions has similar checks not guarded by `#if ASSERTIONS`.